frame/dvm/ethereum/src/mock.rs

```diff
@@ -239,6 +240,100 @@ impl darwinia_ethereum::Config for Test {
 	type StateRoot = IntermediateStateRoot;
 }
 
+// --- pallet-bridge-dispatch config start ---
+
+pub type BridgeMessageId = [u8; 4];
+pub type SubChainId = [u8; 4];
+pub const SOURCE_CHAIN_ID: SubChainId = *b"srce";
+pub const TARGET_CHAIN_ID: SubChainId = *b"trgt";
+
+pub struct AccountIdConverter;
+
+impl sp_runtime::traits::Convert<H256, AccountId32> for AccountIdConverter {
+	fn convert(hash: H256) -> AccountId32 {
+		AccountId32::new(hash.0)
+	}
+}
+
+#[derive(Decode, Encode)]
+pub struct EncodedCall(pub Vec<u8>);
+
+impl From<EncodedCall> for Result<Call, ()> {
+	fn from(call: EncodedCall) -> Result<Call, ()> {
+		Call::decode(&mut &call.0[..]).map_err(drop)
+	}
+}
+
+pub struct CallFilter;
+impl CallFilterT<Origin, Call> for CallFilter {
```

Comment:
As we have discussed before, we want to validate the transaction via `Ethereum::validate_transaction_in_block(id, tx)` in this filter, if false, early return. I find this will always be false in the current implementation as the dispatch origin's balance here is always 0.  The operation of  [pay_dispatch_fee](https://github.com/darwinia-network/darwinia-messages-substrate/blob/main/modules/dispatch/src/lib.rs#L316-L331) is behind the [CallFilter](https://github.com/darwinia-network/darwinia-messages-substrate/blob/main/modules/dispatch/src/lib.rs#L278-L288).

Reply:
Move the part of "transfer fees from relayer to dispatch account" this following `pre_dispatch` trait impl.